session/session.go

```diff
@@ -3,16 +3,35 @@ package session
 import (
 	"crypto/aes"
 	"crypto/cipher"
+	"crypto/hmac"
 	"crypto/sha1"
 	"encoding/base64"
+	"encoding/hex"
+	"errors"
 	"net/url"
 	"strings"
 
 	"golang.org/x/crypto/pbkdf2"
 )
 
 func generateSecret(base, salt string) []byte {
-	return pbkdf2.Key([]byte(base), []byte(salt), key_iter_num, key_size, sha1.New)
+	return pbkdf2.Key([]byte(base), []byte(salt), keyIterNum, keySize, sha1.New)
+}
+
+// I stolen the snippet from: https://gist.github.com/doitian/2a89dc9e4372e55335c9111f576b47bf
+func verifySign(encryptedData, sign, base, signSalt string) bool {
+	signKey := generateSecret(base, signSalt)
+	signHmac := hmac.New(sha1.New, signKey)
+	signHmac.Write([]byte(encryptedData))
+	verifySign := signHmac.Sum(nil)
+	signDecoded, err := hex.DecodeString(sign)
+	if err != nil {
+		return false
```

Comment:
I'm not really sure if an error should be silently dropped here. Did you consider returning it to the caller?